File: DataStructure/Node js/DynamicQueue.js
class Queue {
  constructor(value) {
    this.value = [];
    this.size = 0;
    this.bootom = 0;
  }
  isEmpty() {
    return this.size === 0;
  }

  enqueue(data) {
    this.value[this.size] = data;
    this.size++;
  }
  dequeue() {
    let result = this.value[this.bootom];
    this.value[this.bootom] = null;
    this.size--;
    return result;

  }
  peek() {
    return this.value[this.bootom];
  }
  print() {
    for (let i = 0; i <= this.size; i++) {
      console.log(this.value[i]);
    }
  }

}

let qq = new Queue();
qq.enqueue(1);
qq.enqueue(2);
qq.enqueue(3);
qq.enqueue(4);
qq.enqueue(5);
qq.enqueue(6);

qq.print();
console.log("after dequeue opperation");
qq.dequeue();
qq.print();

File: DataStructure/src/DynamicQueue.java

public class DynamicQueue {

	public int maxSize;

	public int[] queue;
	private int bootom = 0;

	private int top;

	public DynamicQueue(int size) {
		this.maxSize = size;
		queue = new int[maxSize];
		top = -1;
	}

	public void enqueue(int value) {
		if (!isFull()) {
			top++;
			queue[top] = value;
		} else {
			resize(maxSize * 2);
			enqueue(value);
		}
	}

	public int dequeue() {
		if (!isEmpty()) {
			return queue[bootom];
		}

		if (top < maxSize / 4) {
			resize(maxSize / 2);
			return dequeue();
		} else {
			System.out.println("The stack is already empty");
			return -1;
		}
	}

	public int peek() {
		if (!isEmpty())
			return queue[bootom];

		else {
			System.out.println("stack is empty");
			return -1;
		}
	}

	private void resize(int newSize) {
		int[] transferArray = new int[newSize];

		for (int i = 0; i < queue.length; i++) {
			transferArray[i] = queue[i];
			queue = transferArray;
		}
		maxSize = newSize;
	}

	public boolean isEmpty() {
		return (top == -1);
	}

	public boolean isFull() {
		return (top + 1 == maxSize);
	}
	public void print() {
		for(int i = 0; i<queue.length; i++) {
			System.out.println(queue[i]);
		}

}}
